src/Tools/dotnet-monitor/Egress/AzureBlob/AzureBlobEgressProvider.cs

```diff
@@ -143,6 +166,19 @@ public AzureBlobEgressProvider(ILogger<AzureBlobEgressProvider> logger)
             }
         }
 
+        public void AddConfiguredMetadata(AzureBlobEgressProviderOptions options, EgressArtifactSettings artifactSettings)
+        {
+            if (options.IncludeHostNameAsMetadata)
+            {
+                artifactSettings.Metadata.Add(HostNameMetadataKey, Dns.GetHostName());
+            }
+
+            foreach (var metadataPair in options.Metadata)
+            {
+                artifactSettings.CustomMetadata.Add(metadataPair.Key, Environment.GetEnvironmentVariable(metadataPair.Value));
```

Comment:
This will throw if the custom metadata contains a key that matches one of the built in ones e.g. hostname.

Reply:
Are you sure? I intentionally am adding to `CustomMetadata` instead of `Metadata`, which I created mostly for this purpose.

Reply:
If someone adds `"HostName": "..."` as an entry in the `Metadata` property in configuration, then this is going to throw, correct?

Reply:
It shouldn't - the throw will happen around L130 when we try to merge `CustomMetadata` and `Metadata`, hence catching `ArgumentException` in the current implementation.

Reply:
Sorry, another wrong reading on my part. You are adding the hostname metadata to `artifactSettings.Metadata` whereas the line is adding each `options.Metadata` to `artifactSettings.CustomMetadata`.

Is there a reason you are not adding the custom metadata directly to `artifactSettings.Metadata` here, which then alleviates the need for the rest of the code to understand that there is custom metadata?

Reply:
That was what I did originally, but then if there was an issue with some of the custom metadata, I couldn't add any of the metadata (since the good built-in metadata was intermingled with the contaminated custom metadata). Keeping them separate allows us to add the built-in metadata if the merged metadata fails. However, if we actually go validate each item individually (which would also require manually checking if the custom metadata names are legal), we could just put everything in one big dictionary and yank out anything that doesn't belong in there.